### DISCUSSION TITLE
How to filter (one or more) specific accessor(s)/column(s) from outside of table?

### DISCUSSION THREAD

**Discussion by nuri-engin** [TanStack/table#3084] [Q&A]
Hello there, 

I kept looking all around on here, spectrum.chat, StackOverflow, and around the web to find a proper answer/way but couldn't find it out. Thus I'll ask my question here;

I have got a challenge to _filter a couple of different accessors/columns from a single dropdown menu item(s) selection_. 
I'll kindly ask you to check out this  [sample screen-shot](https://nimb.ws/DTN3d7), thus I can explain the need in a better way!

As you'll notice on the screenshot; 
A: I need to filter 1+ accessor/columns **_(but not all entire table!)_**, eg: `User ID`, `Reserved` and `Signal` columns.
B: So the dropdown selection button _(named as "Type")_ will contain those pre-defined filters within menu-items, eg: Gps will filter the `Signal` column and the user may also select another item to filter another column.

So as you see, I need to use the power and logic of [`useFilter`](https://react-table.tanstack.com/docs/api/useFilters
) outside of the table for multiple pre-defined columns.
I was thinking that [`useGlobalFilter`](https://react-table.tanstack.com/docs/api/useGlobalFilter#useglobalfilter) would be a solution but it works only for entire-table filtering.

I'll kindly ask you for any explanation, sample for `functional-component` based approach.

Thanks in advance.

**Comment by nuri-engin** [comment]
I suppose no idea exists for such a need? :(

**Reply by nuri-engin** [reply]
No feed-back yet :(

**Comment by tychenjiajun** [comment]
pass the setFilter method to that dropdown component
